### PR TITLE
[WICKET-6750] allow to abort currently executing request for a given …

### PR DESCRIPTION
**Pull request by reiern70**
allow to abort currently executing request for a given Ajax channel + example. It uses suggestion by @svenmeier. If these changes are not dimmed necessary or desirable I can create some sample project somewhere illustrating how to do this.

### CODE REVIEW

File: wicket-core/src/main/java/org/apache/wicket/ajax/res/js/wicket-ajax-jquery.js [hunk 1]
```diff
@@ -335,6 +335,58 @@
 
 	Wicket.ChannelManager.FunctionsExecuter = FunctionsExecuter;
 
+	/**
+	 * Channel manager maintains a map of channels.
```
Comment (reiern70) [inline wicket-core/src/main/java/org/apache/wicket/ajax/res/js/wicket-ajax-jquery.js]: Change this.

File: wicket-core/src/main/java/org/apache/wicket/ajax/res/js/wicket-ajax-jquery.js [hunk 2]
```diff
@@ -335,6 +335,58 @@
...
+			);
+
+			Wicket.Event.subscribe('/ajax/call/complete',
+				function(jqEvent, attributes, jqXHR, errorThrown, textStatus) {
+					delete requests[attributes.ch];
+				}
+			);
+		},
+
+		getReuest: function (channel) {
```
Comment (reiern70) [inline wicket-core/src/main/java/org/apache/wicket/ajax/res/js/wicket-ajax-jquery.js]: and this

**Comment by svenmeier** [comment]
How about extracting that code into a behavior, e.g. something which can be added to a button?
This would be much nicer to use (showing/hiding the button, keeping track of the currrent Ajax request, cancelling it).
With its own JS file it we could keep this stuff out of the core JS, maybe add this feature to wicket-extensions.

**Comment by reiern70** [comment]
> How about extracting that code into a behavior, e.g. something which can be added to a button?
> This would be much nicer to use (showing/hiding the button, keeping track of the currrent Ajax request, cancelling it).
> With its own JS file we could keep this stuff out of the core JS, maybe add this feature to wicket-extensions.

@svenmeier Many thanks for feedback. This sounds like a very good idea. I will do that and create a new PR.

**Pull request closed without merging**